Description: Properly hide private symbols in libldb1
Author: Jelmer Vernooij <jelmer@debian.org>
Bug: https://bugzilla.samba.org/show_bug.cgi?id=9357
Status: submitted upstream

diff -ur ldb-1.1.13/buildtools/wafsamba/samba_abi.py ldb-1.1.13-fixd/buildtools/wafsamba/samba_abi.py
--- ldb-1.1.13/buildtools/wafsamba/samba_abi.py	2012-09-07 16:47:16.000000000 +0200
+++ ldb-1.1.13-fixd/buildtools/wafsamba/samba_abi.py	2012-11-05 19:23:40.010206344 +0100
@@ -190,12 +190,12 @@
             f.write("\t\t%s;\n" % x)
     else:
         f.write("\t\t*;\n")
-    if len(local_abi) > 0:
+    if abi_match != ["*"]:
         f.write("\tlocal:\n")
         for x in local_abi:
             f.write("\t\t%s;\n" % x[1:])
-    elif abi_match != ["*"]:
-        f.write("\tlocal: *;\n")
+        if len(global_abi) > 0:
+            f.write("\t\t*;\n")
     f.write("};\n")
     f.close()
 
